src/simplify_algebra.cpp

```diff
@@ -453,6 +491,23 @@ struct find_splits
             if(op.name() == "slice")
                 continue;
 
+            if(contains(op.name(), "reduce"))
```

Comment:
If we are going to support reduce then the matcher should be updated to match reduce.